toplevel/role/role.go

```diff
@@ -83,6 +104,19 @@ func (e entry) Save() error {
 	return nil
 }
 
+// Helper function to compare two maps
+func isEqual(existing, new map[string]interface{}) bool {
```

Comment:
This method's function is similar in purpose to https://github.com/app-sre/vault-manager/blob/master/pkg/vault/reconcile.go#L29